docs/sources/reference/glossary.md

```diff
@@ -0,0 +1,194 @@
+does not have state and it never changes.
+
+## libcontainer
+
+libcontainer provides a native Go implementation for creating containers with
+namespaces, cgroups, capabilities, and filesystem access controls. It allows
+you to manage the lifecycle of the container performing additional operations
+after the container is created.
+
+## link
+
+links provide an interface to connect docker containers running on the same host
+to each other without exposing the hosts' network ports. When you set up a link,
+you create a conduit between a source container and a recipient container.
+The recipient can then access select data about the source. To create a link,
+you can use the `--link` flag.
+
+## machine
+
+machine is a docker tool which makes it really easy to create docker hosts on 
+your computer, on cloud providers and inside your own data center. It creates
+servers, installs Docker on them, then configures the docker client to talk to them.
+
+*Also known as : docker-machine*
+
+## overlay
+
+OverlayFS is a [filesystem](#filesystem) service for Linux which implements a
+[union mount](http://en.wikipedia.org/wiki/Union_mount) for other file systems.
+It is supported by the docker daemon as a storage driver.
+
+## registry
+
+A Registry is a hosted service containing [repositories](#repository) of [images](#image)
+which responds to the Registry API.
+
+The default registry can be accessed using a browser at [Docker Hub](#Docker-Hub)
+or using the `docker search` command.
+
+## repository
+
+A repository is a set of docker images. A repository can be shared by pushing it
+to a [registry](#registry) server. The different images in the repository can be
+labeled using [tags](#tag). 
+
+Here is an example of the shared [nginx repository](https://registry.hub.docker.com/_/nginx/)
+and its [tags](https://registry.hub.docker.com/_/nginx/tags/manage/)
+
+## swarm
+
+swarm is a native clustering tool for docker. Swarm pools together several docker hosts
+and exposes them as a single virtual docker host. It serves the standard Docker API,
+so any tool that already works with Docker can now transparently scale up to multiple hosts.
+
+*Also known as : docker-swarm*
+
+## tag
+
+A tag is a label applied to a docker image in a [repository](#repository).
+tags are how various images in a repository are distinguished from each other.
+
+## Union file system
+
+Union file systems, or UnionFS, are file systems that operate by creating layers, making them
+very lightweight and fast. Docker uses union file systems to provide the building 
+blocks for containers.
+
+
+## Virtual Machine
+
+A virtual machine is a program emulates a complete computer and imitates dedicated hardware. 
+It shares physical hardware resources with other users but isolates the operating system.The
+end user has the same experience on a virtual machine as they would have on dedicated hardware.
+
+As compared to containers, it is heavier to run, provides more isolation,
```

Comment:
As compared to -> Compared to ?

"it is heavier to run" -> "a Virtual Machine is"?